src/devices/bus/ieee488/hp9133.cpp

```diff
@@ -0,0 +1,878 @@
+	// 1
+	// 0 - FDCRst#
+	LOG("%s: %02x\n", __func__, data);
+	if (!(data & IO2_W_CLRIND_N))
+		m_floppy_index_int = false;
+	if (data & IO2_W_FDMOTON_N) {
+		m_motor_timer->adjust(attotime::from_msec(2000));
+	} else {
+		m_motor_timer->reset();
+		floppy->mon_w(0);
+	}
+	if (!(data & IO2_W_DKCHRES_N))
+		m_floppy->get_device()->dskchg_w(false);
+	m_floppy->get_device()->ss_w((data & IO2_W_FDHDSEL_N) ? 0 : 1);
+	if (!(data & IO2_W_FDCRES_N))
+		m_fdc->reset();
+}
+
+void hp9133_device::io3_w(uint8_t data)
+{
+	// 7 - DMA enable
+	// 6 - 1 GPIPACK, 0 FDCACK
+	// 5 - Write Precomp?
+	// 4 - FAST enable?
+	// 3 -
+	// 2 - Interrupt Enable
+	// 0:1 - Interrupt select
+
+	m_dmaenable = data & IO3_W_DMA;
+	m_dmaack_switch = data & IO3_W_HPB_FDC_N;
+	m_dma_rwn = data & IO3_W_DMAR_W_N;
+	m_intenable = !(data & IO3_W_INTENBL_N);
+	m_fast = data & IO3_W_FAST;
+	m_intsel = data & IO3_W_INTSEL_MASK;
+
+	LOG("%s: %02x = INTSEL %d, INT %d DMA %d DMA start %d DMA ACK %s\n", __func__,
+	    data, m_intsel, m_intenable, m_dmaenable, (data >> 4) & 1,
+	    m_dmaack_switch ? "GPIB" : "FDC");
+	update_intsel();
+	if (m_fast)
+		m_fast_timer->adjust(attotime::from_usec(1000));
+	else
+		m_fast_timer->adjust(attotime::never);
+}
+
+uint8_t hp9133_device::io1_r(void)
+{
+	uint8_t ret;
+
+	// 7 - hdindex
+	// 6 - HDC int
+	// 5 - HDC DRQ
+	// 4 - HDC BCS
+	// 3 - 0 Ident
+	ret = m_model->read() ^ 0xf;
+	if (m_harddisk_index)
+		ret |= IO1_R_HDIND;
+	if (m_hdc_intrq)
+		ret |= IO1_R_HDINT;
+	if (m_hdc_bdrq)
+		ret |= IO1_R_BDRQ;
+	if (m_hdc_bcs)
+		ret |= IO1_R_BCS_N;
+	LOG("%s: %02x\n", __func__, ret);
+	return ret;
+}
+
+uint8_t hp9133_device::io2_r(void)
+{
+	uint8_t ret = 0;
+
+	// 7 - Write Fault Latch
+	// 5 - FlopDC#
+	// 4 - FDC interrupt
+	// 3 - 0 GPIB Address
+	if (machine().side_effects_disabled())
+		return 0;
+
+	if (m_selftesten->read()) {
+		ret = (m_selftest->read() & 7) ^ 0xf;
+	} else {
+		ret = m_hpib_addr->read() ^ 0xf;
+	}
+	if (m_fdc_irq)
+		ret |= IO2_R_FDCINT;
+	if (!m_floppy->get_device()->dskchg_r())
+		ret |= IO2_R_FDKDCH_N;
+	LOG("%s: %02x\n", __func__, ret);
+	return ret;
+}
+
+uint8_t hp9133_device::io3_r(void)
+{
+	uint8_t ret = 0;
+
+	// 7 - self test enable
+	// 6 - self test MSB
+	// 5 - FASTDON
+	// 3-0 Volume switch
+	ret = m_volcfg->read() ^ 0xf;
+	if (!m_selftesten->read()) {
+		ret |= IO3_R_TEST;
+	} else {
+		if (m_selftest->read() & 8)
+			ret |= IO3_R_SELTST;
+	}
+
+	if (m_dma_addr >= 256)
+		ret |= IO3_R_FASTDON;
+	LOG("%s: %02x\n", __func__, ret);
+	return ret;
+}
+
+void hp9133_device::dma_ram_w(uint8_t data)
+{
+	LOG("%s: %03x = %02x\n", __func__, m_dma_addr, data);
+	if (m_dma_addr < sizeof(m_dma_ram))
+		m_dma_ram[m_dma_addr++] = data;
+}
+
+uint8_t hp9133_device::dma_ram_r(offs_t offset)
+{
+	uint8_t ret;
+
+	if (machine().side_effects_disabled())
+		return 0;
+	if (m_dma_addr >= sizeof(m_dma_ram))
+		return 0;
+	ret = m_dma_ram[m_dma_addr++];
+	LOG("%s: %03x = %02x\n", __func__, m_dma_addr - 1, ret);
+	return ret;
+}
```

Comment:
You still want to return the appropriate value here for the debugger – you only want to suppress the address increment.  Something like:
```
	if (m_dma_addr >= sizeof(m_dma_ram))
		return 0;
	uint8_tconst ret = m_dma_ram[m_dma_addr];
	if (!machine().side_effects_disabled())
	{
		LOG("%s: %03x = %02x\n", __func__, m_dma_addr, ret);
		m_dma_addr++;
	}
	return ret;
```

Reply:
Ok.